### PR TITLE
Bom 1021

### PR DESCRIPTION
**Pull request by Ayub-Khan**
-updated tox and travis
-show deprication warnings

### CODE REVIEW

File: test_settings.py [hunk 1]
```diff
@@ -24,10 +24,9 @@
     'provider.oauth2',
 )
 
-MIDDLEWARE_CLASSES = (
+MIDDLEWARE = (
     'django.contrib.sessions.middleware.SessionMiddleware',
     'django.contrib.auth.middleware.AuthenticationMiddleware',
-    'django.contrib.auth.middleware.SessionAuthenticationMiddleware',
```
Comment (awais786) [inline test_settings.py]: Session verification is enabled and mandatory in Django 1.10 (there’s no way to disable it) regardless of whether or not SessionAuthenticationMiddleware is enabled. In older versions, this protection only applies if django.contrib.auth.middleware.SessionAuthenticationMiddleware is enabled in MIDDLEWARE.

Comment (Ayub-Khan) [reply test_settings.py]: https://docs.djangoproject.com/en/2.2/releases/2.0/#miscellaneous

**Comment by Ayub-Khan** [comment]
https://github.com/edx/edx-drf-extensions/pull/96

**Comment by Ayub-Khan** [comment]
https://github.com/edx/django-oauth2-provider/pull/63

**Comment by Ayub-Khan** [comment]
https://github.com/edx/django-oauth2-provider/pull/63#issuecomment-552728166

File: test_settings.py [hunk 2]
```diff
@@ -14,6 +14,7 @@
 }
 
 INSTALLED_APPS = (
+    'django.contrib.admin',
```
Comment (nasthagiri) [inline test_settings.py]: Q: Why was this additional library needed?

Comment (dianakhuang) [reply test_settings.py]: I believe this was implicitly added to `INSTALLED_APPS` before Django 2.1 and then needed to be explicitly added after that. (aka, tests failed and this is what the internet told me what I needed to do to fix it.)

File: tox.ini [hunk 3]
```diff
@@ -1,5 +1,6 @@
 [tox]
-envlist = py36-django111-drf{37,38,39,latest}, py36-django{20,21}-drf{latest}, quality
+envlist = py{27,35}-django111-drf{36,37,38,39,latest}, py35-django{20,21,22}-drf{latest}, quality
```
[the remark this answers is not in the document]
Comment (dianakhuang) [reply tox.ini]: I believe these tox environments are leftover from previous DRF upgrade work. I can remove the old versions.

**Pull request merged**